62/serfer2/pycon_es.py

```diff
@@ -0,0 +1,155 @@
+# coding=utf-8 -*-
+
+import json
+import requests
+import sys
+import pandas
+import matplotlib.pyplot as plt
+
+from scraper import NameScraper
+
+GENDER_API_KEY = 'ERTTRyulCoHJXVhUEF'
+NAMES_CACHE_FNAME = 'names.csv'
+RESULTS_FNAME = 'results.csv'
+PYCON_ES_SOURCES = (
+    {
+        'year': '2018',
+        'url': 'https://2018.es.pycon.org/#schedule',
+        'xpath': '//div[@class="schedule__talk--speaker"]/div/p/text()'
+    }, {
+        'year': '2017',
+        'url': 'https://2017.es.pycon.org/es/schedule/',
+        'xpath': '//div[@class="slot-inner"]/p/strong/text()'
+    }, {
+        'year': '2016',
+        'url': 'http://2016.es.pycon.org/es/schedule/',
+        'xpath': '//div[@class="slot-inner"]/p/strong/text()'
+    }, {
+        'year': '2015',
+        'url': 'http://2015.es.pycon.org/es/schedule/',
+        'xpath': '//div[@class="slot-inner"]/p/strong/text()'
+    }, {
+        'year': '2014',
+        'url': 'http://2014.es.pycon.org/talks',
+        'xpath': '//div[@id="onlytext"]/h1[@class="text-center"]/text()'
+    },
+    {
+        'year': '2013',
+        'url': 'http://2013.es.pycon.org/#agenda',
+        'xpath': '//ul[@class="schedule-list"]/li/div/a[@target="_blank"]/text()'
+    }
+)
+KNOWN_GENDERS = ['male', 'female', 'unknown']
+
+
+def get_gender(name):
+    """
+    Gets name gender from https://gender-api.com
+    """
+    name_data = json.loads(requests.get(f'https://gender-api.com/get?name={name}&key={GENDER_API_KEY}').content)
+    return name_data.get('gender', '')
+
+
+def load_names_cache():
+    """
+    Read local files with names by gender
+    """
+    names = dict()
+    with open(NAMES_CACHE_FNAME, 'r') as fi:
+        lines = fi.readlines()
+        for line in lines:
+            line = line.strip()
+            if not line:
+                continue
+            name, gender = line.split(';')
+            names[name] = gender
+    return names
+
+
+def save_names_cache(names):
+    """
+    Saves names and gendes in CSV
+    """
+    with open(NAMES_CACHE_FNAME, 'w') as fo:
+        for name, gender in names.items():
+            fo.write(f'{name};{gender}\n')
+
+
+def save_results(results):
+    """
+    saves resulting data to CSV
+    """
+    with open(RESULTS_FNAME, 'w') as fo:
+        fo.write('year,total,male,male %,female,female %\n')
```

Comment:
dont use 'male %' as a column name. It does work with pandas but only when passing the name as string. However, '%' is a very special char and should not be used in a column name that will be referenced often and maybe in different languages or packages. I suggest only use "_" as word delimiter and no special chars like "%". So much better would be your alternative 'male_pcnt' which by the way can be referenced in pandas with `df.male_pcnt'